driver-core/src/test/unit/com/mongodb/client/model/search/SearchOperatorTest.java

```diff
@@ -581,6 +591,96 @@ void near() {
         );
     }
 
+    @Test
+    void in() {
+        ObjectId objectId = new ObjectId();
+        UUID uuid = UUID.randomUUID();
+        assertAll(
+                () -> assertThrows(IllegalArgumentException.class, () ->
+                        // paths must not be empty
+                        SearchOperator.in(null, true)
+                ),
+                () -> assertThrows(IllegalArgumentException.class, () ->
+                        // values should be of one supported BSON types and can't be a mix of different types
+                        SearchOperator.in(fieldPath("fieldName1"), asList(true, "value"))
+                ),
```

Comment:
This test does not pass. It can't, unless we inspect every element of the `values` argument at run time, which no longer happens as of 65840ebcafc785839823775eda11a8d11dd2ea1c (I don't think we should do such a check).